retrieval/target.go

```diff
@@ -235,6 +239,58 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	t.metricRelabelConfigs = cfg.MetricRelabelConfigs
 }
 
+func newHTTPClient(cfg *config.ScrapeConfig) *http.Client {
+	tlsConfig := &tls.Config{}
+
+	// If a CA cert is provided then let's read it in so we can validate the
+	// scrape target's certificate properly.
+	if len(cfg.CACert) > 0 {
+		caCertPool := x509.NewCertPool()
+		// Load CA cert.
+		caCert, err := ioutil.ReadFile(cfg.CACert)
+		if err != nil {
+			log.Errorf("Unable to use specified CA cert %s: %s", cfg.CACert, err)
```

Comment:
Unsure whether logging and continuing is the right course of action here.
I'd prefer returning `(*http.Client, error)` and log and abort in `Update()`.